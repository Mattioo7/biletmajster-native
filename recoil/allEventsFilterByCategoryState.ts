import { atom } from 'recoil';

const allEventsFilterByCategoryState = atom<number | undefined>({
	key: 'allEventsFilterByCategoryState',
	default: undefined
});

export default allEventsFilterByCategoryState;
