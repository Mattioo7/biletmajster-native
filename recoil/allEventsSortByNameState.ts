import { atom } from "recoil";

const allEventsSearchNameState = atom<string>({
  key: "allEventsSearchNameState",
  default: "",
});

export default allEventsSearchNameState;
