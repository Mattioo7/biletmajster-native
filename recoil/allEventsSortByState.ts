import { atom } from 'recoil';

const eventSortByLabels = {
	name: "Name",
	freePlace: "Free place",
	startTime: "Start time",
	endTime: "End time"
};

export type EventSortBy = keyof typeof eventSortByLabels;

export const eventSortByPairs: { value: EventSortBy, label: string }[] = Object.keys(eventSortByLabels).map(key => { return { value: key, label: eventSortByLabels[key] } as any });

const allEventsSortByState = atom<EventSortBy | undefined>({
	key: 'allEventsSortByState',
	default: undefined
});

export default allEventsSortByState;
