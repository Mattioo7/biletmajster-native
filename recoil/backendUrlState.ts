import { atom } from "recoil";

export const urls = [
  {
    name: "BiletMajster",
    url: "https://biletmajster.azurewebsites.net",
  },
  {
    name: "Dionizos",
    url: "https://dionizos-backend-app.azurewebsites.net",
  },
  {
    name: "IO2Central",
    url: "http://io2central-env.eba-vfjwqcev.eu-north-1.elasticbeanstalk.com",
  },
  {
    name: "localhost",
    url: "10.0.2.2:8000",
  },
];

export const backendUrlState = atom<string>({
  key: "backendUrlState",
  default: urls[0].url,
});
