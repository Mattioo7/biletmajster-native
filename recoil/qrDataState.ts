import { atom } from 'recoil';

export type QRData = {
	coreData: string,
	title: string | undefined,
	description: string | undefined
}

const qrDataState = atom<QRData>({
	key: 'qrDataState',
	default: {
		coreData: "Empty code",
		title: "Empty code",
		description: "This code is empty."
	}
});

export default qrDataState;
