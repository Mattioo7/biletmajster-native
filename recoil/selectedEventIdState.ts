import { atom } from 'recoil';

const selectedEventIdState = atom<number | undefined>({
	key: 'selectedEventIdState',
	default: undefined
});

export default selectedEventIdState;
