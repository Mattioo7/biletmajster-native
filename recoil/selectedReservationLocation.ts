import { atom } from 'recoil';

const selectedReservationLocation = atom({
	key: 'selectedReservationLocation',
	default: { latitude: 52, longitude: 21 }
});

export default selectedReservationLocation;
