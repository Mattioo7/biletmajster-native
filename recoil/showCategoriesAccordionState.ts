import { atom } from "recoil";

const showCategoriesAccordionState = atom<boolean>({
  key: "showCategoriesAccordionState",
  default: false,
});

export default showCategoriesAccordionState;
