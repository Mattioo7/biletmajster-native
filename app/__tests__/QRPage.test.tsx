import renderer from "react-test-renderer";
import { RecoilRoot } from "recoil";
import ModalScreen from "../QRPage";
import React from "react";

jest.mock("expo-router", () => ({
  useNavigation: jest.fn().mockImplementation(() => {
    return { setOptions: jest.fn() };
  }),
}));

it("renders correctly", () => {
  const tree = renderer
    .create(
      <RecoilRoot>
        <ModalScreen />
      </RecoilRoot>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
