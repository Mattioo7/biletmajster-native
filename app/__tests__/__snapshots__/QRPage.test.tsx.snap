// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "column",
          "gap": 10,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="650"
      bbWidth="650"
      focusable={false}
      height={650}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 650,
            "width": 650,
          },
        ]
      }
      vbHeight={650}
      vbWidth={650}
      width={650}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              [
                0,
                -65536,
                1,
                -16711681,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="grad"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="100%"
          />
        </RNSVGDefs>
        <RNSVGGroup>
          <RNSVGRect
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            height="650"
            propList={
              [
                "fill",
              ]
            }
            width="650"
            x="0"
            y="0"
          />
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGPath
            d="M0 15.476190476190476 L216.66666666666666 15.476190476190476 M309.5238095238095 15.476190476190476 L340.4761904761905 15.476190476190476 M371.42857142857144 15.476190476190476 L402.3809523809524 15.476190476190476 M433.3333333333333 15.476190476190476 L650 15.476190476190476 M0 46.42857142857143 L30.952380952380953 46.42857142857143 M185.71428571428572 46.42857142857143 L216.66666666666666 46.42857142857143 M340.4761904761905 46.42857142857143 L371.42857142857144 46.42857142857143 M433.3333333333333 46.42857142857143 L464.2857142857143 46.42857142857143 M619.047619047619 46.42857142857143 L650 46.42857142857143 M0 77.38095238095238 L30.952380952380953 77.38095238095238 M61.904761904761905 77.38095238095238 L154.76190476190476 77.38095238095238 M185.71428571428572 77.38095238095238 L216.66666666666666 77.38095238095238 M278.57142857142856 77.38095238095238 L309.5238095238095 77.38095238095238 M340.4761904761905 77.38095238095238 L402.3809523809524 77.38095238095238 M433.3333333333333 77.38095238095238 L464.2857142857143 77.38095238095238 M495.23809523809524 77.38095238095238 L588.0952380952381 77.38095238095238 M619.047619047619 77.38095238095238 L650 77.38095238095238 M0 108.33333333333334 L30.952380952380953 108.33333333333334 M61.904761904761905 108.33333333333334 L154.76190476190476 108.33333333333334 M185.71428571428572 108.33333333333334 L216.66666666666666 108.33333333333334 M278.57142857142856 108.33333333333334 L309.5238095238095 108.33333333333334 M340.4761904761905 108.33333333333334 L402.3809523809524 108.33333333333334 M433.3333333333333 108.33333333333334 L464.2857142857143 108.33333333333334 M495.23809523809524 108.33333333333334 L588.0952380952381 108.33333333333334 M619.047619047619 108.33333333333334 L650 108.33333333333334 M0 139.28571428571428 L30.952380952380953 139.28571428571428 M61.904761904761905 139.28571428571428 L154.76190476190476 139.28571428571428 M185.71428571428572 139.28571428571428 L216.66666666666666 139.28571428571428 M278.57142857142856 139.28571428571428 L309.5238095238095 139.28571428571428 M340.4761904761905 139.28571428571428 L402.3809523809524 139.28571428571428 M433.3333333333333 139.28571428571428 L464.2857142857143 139.28571428571428 M495.23809523809524 139.28571428571428 L588.0952380952381 139.28571428571428 M619.047619047619 139.28571428571428 L650 139.28571428571428 M0 170.23809523809524 L30.952380952380953 170.23809523809524 M185.71428571428572 170.23809523809524 L216.66666666666666 170.23809523809524 M247.61904761904762 170.23809523809524 L340.4761904761905 170.23809523809524 M433.3333333333333 170.23809523809524 L464.2857142857143 170.23809523809524 M619.047619047619 170.23809523809524 L650 170.23809523809524 M0 201.1904761904762 L216.66666666666666 201.1904761904762 M247.61904761904762 201.1904761904762 L278.57142857142856 201.1904761904762 M309.5238095238095 201.1904761904762 L340.4761904761905 201.1904761904762 M371.42857142857144 201.1904761904762 L402.3809523809524 201.1904761904762 M433.3333333333333 201.1904761904762 L650 201.1904761904762 M278.57142857142856 232.14285714285714 L340.4761904761905 232.14285714285714 M0 263.0952380952381 L30.952380952380953 263.0952380952381 M92.85714285714286 263.0952380952381 L123.80952380952381 263.0952380952381 M154.76190476190476 263.0952380952381 L216.66666666666666 263.0952380952381 M247.61904761904762 263.0952380952381 L278.57142857142856 263.0952380952381 M309.5238095238095 263.0952380952381 L433.3333333333333 263.0952380952381 M464.2857142857143 263.0952380952381 L495.23809523809524 263.0952380952381 M61.904761904761905 294.04761904761904 L185.71428571428572 294.04761904761904 M216.66666666666666 294.04761904761904 L247.61904761904762 294.04761904761904 M278.57142857142856 294.04761904761904 L309.5238095238095 294.04761904761904 M340.4761904761905 294.04761904761904 L371.42857142857144 294.04761904761904 M433.3333333333333 294.04761904761904 L526.1904761904761 294.04761904761904 M557.1428571428571 294.04761904761904 L650 294.04761904761904 M0 325 L92.85714285714286 325 M185.71428571428572 325 L247.61904761904762 325 M278.57142857142856 325 L309.5238095238095 325 M340.4761904761905 325 L371.42857142857144 325 M464.2857142857143 325 L495.23809523809524 325 M526.1904761904761 325 L588.0952380952381 325 M619.047619047619 325 L650 325 M30.952380952380953 355.95238095238096 L61.904761904761905 355.95238095238096 M92.85714285714286 355.95238095238096 L185.71428571428572 355.95238095238096 M216.66666666666666 355.95238095238096 L247.61904761904762 355.95238095238096 M371.42857142857144 355.95238095238096 L402.3809523809524 355.95238095238096 M433.3333333333333 355.95238095238096 L464.2857142857143 355.95238095238096 M495.23809523809524 355.95238095238096 L557.1428571428571 355.95238095238096 M619.047619047619 355.95238095238096 L650 355.95238095238096 M0 386.9047619047619 L61.904761904761905 386.9047619047619 M123.80952380952381 386.9047619047619 L154.76190476190476 386.9047619047619 M185.71428571428572 386.9047619047619 L216.66666666666666 386.9047619047619 M247.61904761904762 386.9047619047619 L278.57142857142856 386.9047619047619 M309.5238095238095 386.9047619047619 L371.42857142857144 386.9047619047619 M495.23809523809524 386.9047619047619 L557.1428571428571 386.9047619047619 M588.0952380952381 386.9047619047619 L650 386.9047619047619 M247.61904761904762 417.8571428571429 L309.5238095238095 417.8571428571429 M402.3809523809524 417.8571428571429 L433.3333333333333 417.8571428571429 M464.2857142857143 417.8571428571429 L495.23809523809524 417.8571428571429 M526.1904761904761 417.8571428571429 L557.1428571428571 417.8571428571429 M0 448.8095238095238 L216.66666666666666 448.8095238095238 M340.4761904761905 448.8095238095238 L402.3809523809524 448.8095238095238 M464.2857142857143 448.8095238095238 L619.047619047619 448.8095238095238 M0 479.76190476190476 L30.952380952380953 479.76190476190476 M185.71428571428572 479.76190476190476 L216.66666666666666 479.76190476190476 M247.61904761904762 479.76190476190476 L309.5238095238095 479.76190476190476 M464.2857142857143 479.76190476190476 L495.23809523809524 479.76190476190476 M588.0952380952381 479.76190476190476 L650 479.76190476190476 M0 510.7142857142857 L30.952380952380953 510.7142857142857 M61.904761904761905 510.7142857142857 L154.76190476190476 510.7142857142857 M185.71428571428572 510.7142857142857 L216.66666666666666 510.7142857142857 M278.57142857142856 510.7142857142857 L340.4761904761905 510.7142857142857 M371.42857142857144 510.7142857142857 L433.3333333333333 510.7142857142857 M526.1904761904761 510.7142857142857 L557.1428571428571 510.7142857142857 M588.0952380952381 510.7142857142857 L619.047619047619 510.7142857142857 M0 541.6666666666666 L30.952380952380953 541.6666666666666 M61.904761904761905 541.6666666666666 L154.76190476190476 541.6666666666666 M185.71428571428572 541.6666666666666 L216.66666666666666 541.6666666666666 M247.61904761904762 541.6666666666666 L340.4761904761905 541.6666666666666 M433.3333333333333 541.6666666666666 L557.1428571428571 541.6666666666666 M588.0952380952381 541.6666666666666 L650 541.6666666666666 M0 572.6190476190476 L30.952380952380953 572.6190476190476 M61.904761904761905 572.6190476190476 L154.76190476190476 572.6190476190476 M185.71428571428572 572.6190476190476 L216.66666666666666 572.6190476190476 M278.57142857142856 572.6190476190476 L309.5238095238095 572.6190476190476 M340.4761904761905 572.6190476190476 L371.42857142857144 572.6190476190476 M402.3809523809524 572.6190476190476 L464.2857142857143 572.6190476190476 M495.23809523809524 572.6190476190476 L526.1904761904761 572.6190476190476 M557.1428571428571 572.6190476190476 L588.0952380952381 572.6190476190476 M619.047619047619 572.6190476190476 L650 572.6190476190476 M0 603.5714285714286 L30.952380952380953 603.5714285714286 M185.71428571428572 603.5714285714286 L216.66666666666666 603.5714285714286 M278.57142857142856 603.5714285714286 L309.5238095238095 603.5714285714286 M340.4761904761905 603.5714285714286 L402.3809523809524 603.5714285714286 M495.23809523809524 603.5714285714286 L526.1904761904761 603.5714285714286 M0 634.5238095238095 L216.66666666666666 634.5238095238095 M247.61904761904762 634.5238095238095 L309.5238095238095 634.5238095238095 M588.0952380952381 634.5238095238095 L619.047619047619 634.5238095238095 "
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={0}
            strokeWidth="30.952380952380953"
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 24,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Empty code
    </Text>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          undefined,
        ]
      }
    >
      This code is empty.
    </Text>
  </View>
</View>
`;
